tests/API/OVAL/unittests/test_object_component_type.sh

```diff
@@ -2,10 +2,11 @@
 
 stderr=`mktemp`
 
-set -e
 set -o pipefail
 
-$OSCAP oval eval $srcdir/test_object_component_type.oval.xml 2> $stderr || ret=$?
+$OSCAP oval eval $srcdir/test_object_component_type.oval.xml 2> $stderr
+ret=$?
+set -e
```

Comment:
I would suggest to leave `set -e`  where it was and rather change the call to oscap in the following way:

`$OSCAP oval eval $srcdir/test_object_component_type.oval.xml 2> $stderr || ret=$?`

Reply:
It was like that before, but then, the `ret` would not be defined if `oscap` finished without error. My change allows `oscap` to finish with any code it wants, storing it in the variable using a minimal number of steps.

Reply:
IMO this test is expected to fail, if `oscap` finishes without error it is unexpected state in this test (see OVAL file). These constructions with test in `[]` under the command are used in the test suite to explicitly state what is the expected return code (when `set -e` is defined):
`[ $ret -eq 1 ]`

Your change of the code does not change behavior in any way (if `oscap` command passes test fails), it can be merged but the rest of tests are written in a way I described above.

Reply:
The problem is that although it shouldn't, `oscap` can return zero (it actually did, that's how I have discovered this error). Consequently, `[ $ret -eq 1 ]` failed in the original code, but for a wrong reason - because `ret` was not defined. My change ensures that `ret` is always defined using the standard procedure of asking the return code unconditionally after the command ends. As the command is expected to fail, I had to deal with the `set -e` somehow. Putting `&& ret=$? || ret=$?` past the command would make possible to capture zero and any non-zero return codes, but I rather went for moving the `set -e` :-)

Reply:
This is correct and as I said I agree with merging this. Actually it makes code more readable. In future we should rewrite and unify these code constructs in the test library to make the code consistent.